packages/utils/src/internals/robots.ts

```diff
@@ -0,0 +1,80 @@
+// @ts-expect-error This throws a compilation error due to got-scraping being ESM only but we only import types, so its alllll gooooood
+import type { HTTPError as HTTPErrorClass } from 'got-scraping';
+import type { Robot } from 'robots-parser';
+import robotsParser from 'robots-parser';
+
+import { gotScraping } from './gotScraping';
+import { Sitemap } from './sitemap';
+
+let HTTPError: typeof HTTPErrorClass;
+
+export class RobotsFile {
+    private constructor(
+        private robots: Pick<Robot, 'isAllowed' | 'getSitemaps'>,
+        private proxyUrl?: string,
```

Comment:
the signature looks a bit weird this way, i'd prefer either one options parameter (including the proxyUrl) or three separate parameters. but its a minor issue as its private now.